### PR TITLE
adding logical_not to paddle

### LINKED ISSUE #16126: logical_not

### PR DESCRIPTION
**Pull request by ahmadel1**
Fixes #16126

### CODE REVIEW

**Comment by ahmadel1** [comment]
please review this PR @hmahmood24

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_logic.py [hunk 1]
```diff
@@ -269,3 +269,29 @@ def test_paddle_logical_xor(
         x=x[0],
         y=x[1],
     )
+    
+   
+@handle_frontend_test(
+    fn_tree="paddle.logical_not",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"), num_arrays=1, shared_dtype=True
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_logic.py:277]: The `num_arrays` is set to `1` by default so you can remove that. The `shared_dtype` argument is used when we are generating more than `1` arrays so you can remove that as well.

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @ahmadel1. Thanks for making the PR. I've reviewed it and left a few comments for you to address. Additionally, the PR is failing some [lint/formatting checks](https://github.com/unifyai/ivy/actions/runs/5144962653/jobs/9261940735?pr=16145#step:5:103) so can you please fix those as well? Kindly ask for a re-review from me once you're done. Thanks 😃

**Comment by ahmadel1** [comment]
@hmahmood24  Could you please re-review my PR? Your feedback is highly appreciated

**Review by hmahmood24 — APPROVED** [review]

**Comment by hmahmood24** [comment]
Everything looks good to merge now. Thanks so much for the PR @ahmadel1! 🚀

**Pull request merged**